apps/desktop/src/lib/modes/service.ts

```diff
@@ -0,0 +1,36 @@
+import { invoke, listen } from '$lib/backend/ipc';
+import { derived, writable } from 'svelte/store';
+
+type Mode = { type: 'OpenWorkspace' } | { type: 'OutsideWorksapce' } | { type: 'Edit' };
+
+export class ModeService {
+	readonly head = writable<string | undefined>(undefined);
+	readonly mode = writable<Mode | undefined>(undefined);
+
+	readonly gbBranchActive = derived(this.head, (head) => head === 'gitbutler/integration');
+
+	unsubscribe?: () => Promise<void>;
+
+	constructor(private projectId: string) {
+		this.unsubscribe = subscribeToHead(projectId, (head, mode) => {
```

Comment:
Sorry, got in a bit late here. Could we subscribe/unsubscribe using the store like we do in e.g. headService.ts?